tests/test_result.py

```diff
@@ -130,6 +130,14 @@ def test_unwrap_or_else() -> None:
     assert n.unwrap_or_else(str.upper) == 'NAY'
 
 
+def test_unwrap_or_raise() -> None:
+    o = Ok('yay')
+    n = Err('nay')
+    assert o.unwrap_or_raise(lambda e: Exception(e)) == 'yay'
+    with pytest.raises(Exception):
+        n.unwrap_or_raise(lambda e: Exception(e))
```

Comment:
these can be plain `.unwrap_or_raise(Exception)`, in multiple places

while at it, please use `ValueError` or some other more specific exception, to make it clear that the passed in exception type is used for raising (and adjust the `pytest.raises()` call accordingly).

Reply:
updated.